packages/twenty-server/src/engine/api/rest/core/services/rest-api-core.service.ts

```
@@ -98,4 +93,36 @@ export class RestApiCoreService {
       );
     }
   }
+
+  async findDuplicates(request: AuthenticatedRequest) {
+    return await this.restApiFindDuplicatesHandler.handle(request);
+  }
+
+  async update(request: AuthenticatedRequest) {
+    return await this.restApiUpdateOneHandler.handle(request);
+  }
+
+  async delete(request: AuthenticatedRequest) {
+    const { id: recordId } = parseCorePath(request);
+
+    const isCommonApiEnabled = await this.isCommonApiEnabled(request);
+    const isSoftDelete = parseSoftDeleteRestRequest(request);
+
+    if (isCommonApiEnabled && !isSoftDelete && isDefined(recordId))
+      return await this.restApiDestroyOneHandler.commonHandle(request);
+    if (isCommonApiEnabled && !isSoftDelete && !isDefined(recordId))
+      return await this.restApiDestroyManyHandler.handle(request);
```

this will throw later 

```
  async handle(request: AuthenticatedRequest) {
    const { id: recordId } = parseCorePath(request);

    if (!recordId) {
      throw new BadRequestException('Record ID not found');
    }
```
 I think we could early throw, it's a bit confusing at reading to see that an undefined recordId can qualify for a destroy (while actually it cannot and should not)